src/manifests_operator.py

```diff
@@ -8,11 +8,24 @@
 from hashlib import md5
 from typing import Dict
 
-from ops.manifests import ManifestLabel, Manifests
+from lightkube.codecs import AnyResource
+from ops.manifests import ManifestLabel, Manifests, Patch
 
 log = logging.getLogger()
 
 
+class PatchPluginName(Patch):
+    """Adjust name of device plugin from sample naming provided by Intel."""
+
+    def __call__(self, obj: AnyResource) -> None:
+        """Replace sample naming provided by Intel."""
+        if obj.kind == "GpuDevicePlugin":
+            upstream_name: str = obj["metadata"]["name"]
+            if upstream_name.endswith("-sample"):
+                log.info(f"Patching name for {obj.kind} {upstream_name}")
+                obj["metadata"]["name"] = upstream_name.removesuffix("-sample")
```

Comment:
👍🏻

Reply:
Some of the custom classes/types in lightkube are tricky to work with. For example its Generic class wraps a dict under the hood but still exposes attributes that you can read but should not write to (I found that out the hard way).

Reply:
Yeah?  I'm surprised this can't be 


```suggestion
class PatchPluginName(Patch):
    """Adjust name of device plugin from sample naming provided by Intel."""

    def __call__(self, obj: AnyResource) -> None:
        """Replace sample naming provided by Intel."""
        if obj.kind == "GpuDevicePlugin":
            upstream_name: str = obj.metadata.name
            if upstream_name.endswith("-sample"):
                log.info(f"Patching name for {obj.kind} {upstream_name}")
                obj.metadata.name = upstream_name.removesuffix("-sample")
```

Reply:
Then again -- `GpuDevicePlugin` is a CRD right?  Maybe lightkube handler for that is trickier than others

Reply:
Yeah, `GpuDevicePlugin` is a CRD.

I think this is fine because it's reading from the attribute:

```python
upstream_name: str = obj.metadata.name
```

But writing to the attribute (rather than the underlying dict) does not persist the change, I know because this is how I originally had it written. :)

```python
obj.metadata.name = upstream_name.removesuffix("-sample")
```

I think there is some `__getattr__` magic in the class to expose the metadata dict keys as attributes, but it's unclear why writing back to the attribute does not persist the change. There may be some special treatment of the metadata key.